### ISSUE TITLE
Friends System in MTA:SA

### ISSUE THREAD

**Issue by TraderReal** [multitheftauto/mtasa-blue#3054]
### Is your feature request related to a problem? Please describe.

The absence of a friends list in MTA:SA that makes it possible for players to connect and interact with their friends seamlessly. Players have been unable to join their friends directly or determine if their friends are online or offline because of the absence of it.

### Describe the solution you'd like

The solution I suggest is adding a friends list option in MTA:SA that works with IDs or accounts. This feature would allow you to join your friend's server, see if they are online or offline, observe what game they are playing, and even chat with them. Additionally, there would be customizable settings that allow you to control various aspects such as whether people can join you, whether people can see your activity, and whether people can see if you are online.

This functionality would be highly useful and it is common in online games like League of Legends, ROBLOX, and others.

### Describe alternatives you've considered

/

### Additional context

/

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not about a cheat or security vulnerability.

**Comment by Dark-Dragon** [comment]
Hypothetical questions to answer regarding this request:

- Is it worth creating a built-in solution from the ground up within MTA?
- What would the unique identifier look like (community account, forums account, something else entirely) and how would this tie in with server/scripts and regular nick names?
- How would the messaging work? (would offline messages be a thing? would you see them pop up in server-chat?)
- Would it be opt-in, opt-out, outright required?
- Are there privacy/security concerns associated with any of the decisions above?
- What alternatives to creating it for MTA are there? (ie tie in existing friend systems with decent APIs like steam/discord)

**Comment by F2BShady** [comment]
> * What alternatives to creating it for MTA are there? (ie tie in existing friend systems with decent APIs like steam/discord)

A: Perhaps it could be an MTA Forum ID or another type of MTA identifier?

info :We talked about this issue on the MTA development side, for example, the player should watch what server his friend is on or what kind of status he has, if the person allows, he can log in to the server he plays and his friend can log into the server.

**Comment by F2BShady** [comment]
https://i.imgur.com/IO8idRK.png
https://i.imgur.com/tORunrE.png
https://i.imgur.com/eW33GOq.png

I made small designs at the time, I think it can be developed more.

**Comment by PlatinMTA** [comment]
Afaik MTA had an account system back in the day, under the MTA Community. For some reason it was deleted, maybe some admin knows why.

I don't think this is worth it nor needed. I think it would be better to just add a Discord Integration, which will be used more and will also be free advertisement for servers & MTA itself.

**Comment by CarlRayanitch** [comment]
great idea, i love the game to be more modernized as we're stuck in 2013 sadly and servers keep going down a notch.
we can start being better with your suggestion making the menu more full and less autistic

**Comment by ALBANDER2022** [comment]
Yes , it is very useful Feature , We hope that this feature will be available as soon as possible

**Comment by TracerDS** [comment]
With the increased amount of cheaters I think it would be a good idea to implement this.
An account on forum would probably be sufficient.
Making it required will be harsh but it would be worth in the end against cheaters

**Comment by ALBANDER2022** [comment]
> With the increased amount of cheaters I think it would be a good idea to implement this. An account on forum would probably be sufficient. Making it required will be harsh but it would be worth in the end against cheaters

Yes , you are right and we are waiting to apply this feature as soon as possible

**Comment by Fernando-A-Rocha** [comment]
This should come with the remake of the MTA community platform. There should be a central account system and tbh it can be the forums

**Comment by Proxy-99** [comment]
I like this idea

**Comment by PlatinMTA** [comment]
> With the increased amount of cheaters I think it would be a good idea to implement this. An account on forum would probably be sufficient. Making it required will be harsh but it would be worth in the end against cheaters

Required only if the server wants to require it, for sure.
Amh I don't know how this could actually stop cheaters tho*, as I said before I don't think this system is needed still, but it's a nice addition as long as its not required.

\* How can we make sure they aren't just creating multiple accounts? You just need an email and like 2 mins. IP block? Change IP or use a VPN.
